c/dynalloc.c

```diff
@@ -1164,6 +1165,274 @@ int dynallocUnallocDatasetByDDName(const DynallocDDName *ddName,
   return dynallocUnallocDatasetByDDNameInternal(ddName, flags, sysRC, sysRSN, FALSE);
 }
 
+/*************************/
+#define FIELD_MAX_LENGTH    1024
+#define REQUEST_MAX_LENGTH  2048
+typedef int EXTF();
+#pragma linkage(EXTF,OS_UPSTACK)
+
+static int allocDataSetOpenString(dataSetRequest *request, char *buffer,
+                             int bufferSize ) {
+  char * ptr;
+  char field[FIELD_MAX_LENGTH];
+  int  status = 0, value;
+  char *baseRequest = "alloc new catalog msg(2) ";
+  char tempString[20] = {0};
+
+  /* setup initial request string */
+  strncpy(buffer, baseRequest, bufferSize);
+
+  /* Initialize the DD/FI field */
+  ptr = (request->ddName == NULL ?  DATASET_DEFAULT_DDNAME: request->ddName);
+  snprintf(field, sizeof(field)-1, "FI(%s) ", ptr);
+  strncat(buffer, field, bufferSize - strlen(buffer));
+
+  /* Initialize daName */
+  snprintf(field, sizeof(field)-1, "DA(%s) ", request->daName);
+  strncat(buffer, field, bufferSize - strlen(buffer));
+
+  /* Initialize data set type */
+  if ((ptr = request->dsnType) != NULL) {
+    snprintf(field, sizeof(field)-1, "DSNTYPE(%s) ", ptr);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize organization */
+  if (request->organization != NULL) {
+    snprintf(field, sizeof(field)-1, "DSORG(%s) ", request->organization);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize record format */
+  if (request->recordFormat != NULL) {
+    snprintf(field, sizeof(field)-1, "RECFM(%s) ", request->recordFormat);
+    strncat(buffer, field, bufferSize - strlen(buffer));
```

Comment:
if `strlen(buffer)` is greater than `bufferSize`, the result will wrap around, you'll get a huge positive number and a buffer overflow.